### ISSUE TITLE
Use cgroup v2 in rootless mode

### ISSUE THREAD

**Issue by MoetaYuko** [NVIDIA/nvidia-docker#1615]
NVIDIA Container Toolkit doesn't work in rootless mode by default, because cgroup is not supported in rootless mode, disabling its use fixed the issue as mentioned in https://github.com/NVIDIA/nvidia-docker/issues/1155#issuecomment-884499204

However, __limiting resources is supported in rootless mode when running with cgroup v2 and systemd__ according to [docker docs](https://docs.docker.com/engine/security/rootless/#limiting-resources), is it possible to migrate to cgroup v2 so we don't need to set `no-cgroups` in rootless mode?

**Comment by klueska** [comment]
Support for cgroupv2 was added in libnvidia-container 1.8.0.

I've not followed those docs for rootless docker, but rootless podman works with cgroupv2 on the latest libnvidia-container.

**Comment by MoetaYuko** [comment]
> Support for cgroupv2 was added in libnvidia-container 1.8.0.

Yes, it works out-of-box on Debian 11 (which uses cgroupv2) when docker runs as root user, but I have to set `no-cgroups` in `/etc/nvidia-container-runtime/config.toml` to make it work when docker is rootless. Otherwise it errors as https://github.com/NVIDIA/nvidia-docker/issues/1155#issue-536921053 on `docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi`

**Comment by bhack** [comment]
Any news on this?
